api/v1alpha1/envoyproxy_types.go

```diff
@@ -86,8 +86,57 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
+
+	// FilterPriority defines the order of filters in the Envoy proxy's HTTP filter chain.
+	//
+	// +optional
+	FilterPriority []FilterPriority `json:"filterPriority,omitempty"`
+}
+
+// FilterPriority defines the order of filters in the Envoy proxy's HTTP filter chain.
```

Comment:
the API looks good, but lets take a little more time to make sure if we 
* agree on this default order
* should we leave a gap for any upcoming feature like `rbac` or `compression` filter ?

we cannot change these default numbers going forward

Reply:
> should we leave a gap for any upcoming feature like rbac or compression filter or is the 100 gap enough to handle these future cases ?

We could list all the filters that EG is likely to use.